RecCalorimeter/src/components/CorrectCaloClusters.cpp

```diff
@@ -72,21 +77,66 @@ StatusCode CorrectCaloClusters::initialize() {
+      return StatusCode::FAILURE;
+    }
   }
-  if (m_systemIDs.size() != m_downstreamFormulas.size()) {
-    error() <<  "Sizes of systemIDs vector and downstreamFormulas vector does not match, exiting!" << endmsg;
-    return StatusCode::FAILURE;
+  if (m_downstreamCorr) { 
+    if (m_downstreamFormulas.empty() || m_downstreamParams.empty()){ 
+      error() << "Downstream correction is turned on, but downstreamFormulas or downstreamParams vector is empty, exiting!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+    if (m_systemIDs.size() != m_downstreamFormulas.size()) {
+      error() <<  "Sizes of systemIDs vector and downstreamFormulas vector does not match, exiting!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+    if (m_systemIDs.size() != m_downstreamParams.size()) {
+      error() <<  "Sizes of systemIDs vector and downstreamParams vector does not match, exiting!" << endmsg;
+      return StatusCode::FAILURE;
+    }
   }
-  if (m_systemIDs.size() != m_downstreamParams.size()) {
-    error() <<  "Sizes of systemIDs vector and downstreamParams vector does not match, exiting!" << endmsg;
-    return StatusCode::FAILURE;
+  if (m_benchmarkCorr) { 
+    // number of benchmark parameters is 6 
+    size_t nBenchmarkParams = 6; 
+    // check that all benchmark parameters and functions are not empty and have the correct size
+    if (m_benchmarkFormulas.empty() || m_benchmarkParametrization.empty()){ 
+      error() << "Benchmark correction is turned on, but benchmarkFormulas or benchmarkParametrization vector is empty, exiting!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+    if (m_benchmarkFormulas.size() != m_benchmarkParametrization.size()) {
+      error() <<  "Sizes of benchmarkFormulas vector and benchmarkParametrization vector does not match, exiting! " << m_benchmarkFormulas.size() << endmsg;
+      return StatusCode::FAILURE;
+    }
+    if (m_benchmarkParamsApprox.size() != nBenchmarkParams){
+        error() <<  "The benchmarkParamsApprox vector does not have the proper size, the size should be 6." << endmsg;
+        return StatusCode::FAILURE;
+      }
+    if (m_benchmarkEneSwitch<0.){
+      if (m_benchmarkFormulas[0].size() != m_benchmarkParamsApprox.size()){
+        error() <<  "A single formula parametrization for benchmark correction is turned on, but the benchmarkFormulas[0] vector does not have the proper size, the size should be 6." << endmsg;
+        return StatusCode::FAILURE;
+      }
+    }
+    if (m_benchmarkEneSwitch>0.){
+      if (m_benchmarkFormulas.size()<2){
+        error() <<  "The two formulas parametrization for benchmark correction is turned on, but the benchmarkFormulas vector does not have the proper size, the size should be 2." << endmsg;
+        return StatusCode::FAILURE;
+      }
+      if (m_benchmarkFormulas[0].size() != m_benchmarkParamsApprox.size() || m_benchmarkFormulas[1].size() != m_benchmarkParamsApprox.size()){
```

Comment:
shouldn't this be `m_benchmarkFormulas[0]` or `m_benchmarkFormulas[1]`?

Reply:
it could be, but then we would only check that the two vectors have the same size and we would not check that the size is correct. E.g. they both can still have the same size, e.g. 5, while we want 6. While in this case we first check that m_benchmarkParamsApprox has the correct size and use it as a reference later.